### PR TITLE
feat: implement notification and notification stack

### PR DESCRIPTION
**Pull request by TienThinh-fe**
- Add Notification and NotificationStack component
- Create notification store to save notifications state

### CODE REVIEW

File: src/components/Notification/NotificationStack.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      {[...notifications.values()].map((notification) => (
+        <Notification
+          key={notification.id}
+          message={notification.message}
+          type={notification.type}
+          onRemove={() => removeNotification(notification.id)}
+        />
+      ))}
+    </NotificationStackContainer>,
+    container
```
Comment (manng2) [inline src/components/Notification/NotificationStack.tsx:53]: Can we make it to be `<Container/>`?

File: src/pages/TestNotification.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { NotificationType } from '@/libs/constants'
+import { useNotificationStore } from '@/stores/notification'
+
+const TestNotification = () => {
+  const [notifications, setNotifications] = useNotificationStore((state) => [
+    state.notifications,
+    state.setNotifications,
+  ])
+
+  const handleAddNotification = (type: NotificationType, message: string) => {
```
Comment (sonhaaa) [inline src/pages/TestNotification.tsx:11]: This logic should be handle by Notification component by it own. Number of notification that show on screen must be a param. Create a hook (ex: useNotification) and export the addNofication function instead.
ex: `const { addNotification } = useNotification()`

File: src/components/Notification/Notification.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  return (
+    <>
+      {isDisplayed && (
+        <NotificationContainer type={type}>
+          <NotificationContent>
+            <NotificationIcon alt="Info" src={Info} />
+            <NotificationMessage>{message}</NotificationMessage>
+          </NotificationContent>
+          <CloseIcon alt="Close" src={Close} type={type} onClick={onRemove} />
```
Comment (sonhaaa) [inline src/components/Notification/Notification.tsx:87]: Must be wrap by a button tag

File: src/components/Notification/NotificationStack.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import { createContainer } from './container'
+import { Notification } from './Notification'
+
+const NotificationStackContainer = styled.div`
+  position: relative;
+  width: 200px;
+  > div:nth-child(3) {
+    top: 28px;
+    z-index: 3;
```
Comment (sonhaaa) [inline src/components/Notification/NotificationStack.tsx:14]: If I have a element with `position: absolute` and `z-index: 4`, does it overlap with this element?

File: src/components/Notification/Notification.tsx (hunk 3 above)
Comment (TienThinh-fe) [reply src/components/Notification/Notification.tsx]: Please check my third commit, i have pushed the new code. 😎

File: src/components/Notification/NotificationStack.tsx (hunk 4 above)
Comment (sonhaaa) [reply src/components/Notification/NotificationStack.tsx:14]: > the NotificationStack is put in the Container with z-index: 9999, so every component has z-index fewer than 9999 will be overlaid by the Notification. Lemme know if this way is ok

cool

File: src/pages/TestNotification.tsx (hunk 2 above)
Comment (TienThinh-fe) [reply src/pages/TestNotification.tsx]: Please check my forth commit, thanks.

File: src/hooks/useNotification.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { NotificationType } from '@/libs/constants'
+import { useNotificationStore } from '@/stores/notification'
+
+const useAddNotification = () => {
```
Comment (sonhaaa) [inline src/hooks/useNotification.ts:4]: Please use a general noun for naming a hook, for this case, `useNotification `will an okay name.
If we use `useAddNotification `hook, it look like 
`const {addNotification} = useAddNotification()`
look kind of dummy.
Moreover, in some cases, we will need some utility such as `currentNotification`, `isNotification`, `getHistoryNotification`. It's more pro when we hold all the utility function inside that hook. Then, it's gonna be 
`const { addNotification, currentNotification, isNotification, getHistoryNotification } = useNotification()`
It's look more pro 😎

**Review by sonhaaa — CHANGES_REQUESTED** [review]

File: src/components/Notification/NotificationStack.tsx (hunk 4 above)
Comment (manng2) [reply src/components/Notification/NotificationStack.tsx:14]: Does NotificationStack always overlap with other elements? If it does, I suggest you use `2147483647` for `z-index`. 

https://stackoverflow.com/questions/491052/minimum-and-maximum-value-of-z-index

**Comment by TienThinh-fe** [comment]
@manng2 i just pushed new commit about z  index and solving some review of sonha. please give it a review. love u ❤

**Pull request merged**